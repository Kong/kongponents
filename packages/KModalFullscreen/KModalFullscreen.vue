<template>
  <div
    v-if="isVisible"
    :aria-label="title"
    class="k-modal-fullscreen isOpen"
    role="dialog"
    aria-modal="true"
    @keyup.esc="close"
    @keyup.enter="proceed">
    <div
      ref="modalBodyContent"
      class="k-modal-fullscreen-dialog"
      tabindex="0"
    >
      <div class="k-modal-fullscreen-body-header">
        <div
          v-if="$slots['body-header'] || bodyHeader"
          class="body-header">
          <slot name="body-header">{{ bodyHeader }}</slot>
        </div>
        <div
          v-if="$slots['body-header-description'] || bodyHeaderDescription"
          class="body-header-description">
          <slot name="body-header-description">{{ bodyHeaderDescription }}</slot>
        </div>
      </div>

      <div class="k-modal-fullscreen-body">
        <slot name="default" />
      </div>

      <!-- Header at the bottom to allow proper tabindex -->
      <div class="k-modal-fullscreen-header">
        <div
          class="k-modal-fullscreen-header-description"
          role="heading"
          aria-level="2">
          <div class="k-modal-fullscreen-title">
            <span class="header-icon pr-2 my-auto">
              <slot name="header-icon">
                <KIcon :icon="iconString" />
              </slot>
            </span>
            <span class="header-content my-auto">
              <slot name="header-content">{{ title }}</slot>
            </span>
          </div>
          <div class="k-modal-fullscreen-action ml-3">
            <div class="k-modal-fullscreen-action-buttons">
              <slot name="action-buttons">
                <KButton
                  :appearance="cancelButtonAppearance"
                  class="cancel-button"
                  @click="close"
                >
                  {{ cancelButtonText }}
                </KButton>
                <KButton
                  :appearance="actionButtonAppearance"
                  class="proceed-button"
                  @click="proceed"
                >
                  {{ actionButtonText }}
                </KButton>
              </slot>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import KButton from '@kongponents/kbutton/KButton.vue'
import KIcon from '@kongponents/kicon/KIcon.vue'

export default {
  name: 'KModalfullscreen',
  components: { KButton, KIcon },

  props: {
    /**
     * Set the text of the title, if using title slot
     */
    title: {
      type: String,
      required: true
    },
    /**
     * Set the title in the body
     */
    bodyHeader: {
      type: String,
      default: ''
    },
    /**
     * Text to display as a description of the body's title
     */
    bodyHeaderDescription: {
      type: String,
      default: ''
    },
    /**
      *  Pass whether or not the modal should be visible
      */
    isVisible: {
      type: Boolean,
      default: false
    },
    /**
     * Set the text of the close/cancel button
     */
    cancelButtonText: {
      type: String,
      default: 'Cancel'
    },
    /**
     * Set the text of the action/proceed button
     */
    actionButtonText: {
      type: String,
      default: 'Save'
    },
    /**
     * Set the appearance of the action/proceed button
     */
    actionButtonAppearance: {
      type: String,
      default: 'primary'
    },
    /**
     * Set the appearance of the close/cancel button
     */
    cancelButtonAppearance: {
      type: String,
      default: 'outline'
    },
    /**
      *  Pass the type of icon for the header on the left
      */
    iconString: {
      type: String,
      default: 'kong'
    }
  },

  computed: {
    isOpen () {
      return !!this.isVisible
    }
  },

  watch: {
    isVisible: function () {
      if (this.isOpen) {
        document.body.style.overflow = 'hidden'

        this.$nextTick(() => {
          this.$refs.modalBodyContent.focus()
        })
      } else {
        document.body.style.overflow = ''
      }
    }
  },

  mounted () {
    document.addEventListener('keydown', this.handleKeydown)
  },

  beforeDestroy () {
    document.removeEventListener('keydown', this.handleKeydown)
  },
  destroyed () {
    document.body.style.overflow = ''
  },

  methods: {
    handleKeydown (e) {
      if (this.isVisible) {
        if (e.keyCode === 27) { // `esc` key
          this.close()
        } else if (e.keyCode === 13) { // `enter` key
          this.proceed()
        }
      }
    },
    close () {
      this.$emit('canceled')
    },
    proceed () {
      this.$emit('proceed')
    }
  }
}
</script>

<style scoped lang="scss">
@import '~@kongponents/styles/variables';
$screen-sm: 768px;
$screen-md: 992px;
$fullscreen-modal-padding: 64px;

.k-modal-fullscreen-dialog {
  padding-top: $fullscreen-modal-padding * 2;
  background: var(--white);
  z-index: 9999;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100vw;

  @media only screen and (min-width: ($screen-sm + 1px)) {
    padding-top: $fullscreen-modal-padding;
  }
}

.k-modal-fullscreen-header {
  position: fixed;
  display: flex;
  top: 0;
  width: 100%;
  flex-direction: column;
  padding: var(--spacing-lg) 0;
  background-color: var(--white);
  border-bottom: 1px solid var(--grey-300);

  .k-modal-fullscreen-header-description {
    display: flex;
    justify-content: space-between;
    font-size: var(--KModalFullscreenHeaderSize, 20px);
    font-weight: var(--KModalFullscreenHeaderWeight, 500);
    color: var(--KModalFullscreenHeaderColor, var(--black-500, color(black-500)));
  }

  .k-modal-fullscreen-body {
    text-align: center;
    position: relative;
    flex: 1 1 auto;
    font-size: var(--KModalFullscreenFontSize, 13px);
    line-height: 20px;
  }
}

.k-modal-fullscreen-title {
  display: inline-flex;
  position: relative;
  margin-left: 36px;
}

.k-modal-fullscreen-action {
  display: inline-flex;
  margin-right: var(--spacing-xl, spacing(xl));

  & button {
    height: 40px;
    margin-left: var(--spacing-md, spacing(md));
    font-weight: 400;
    font-size: 13px;
    line-height: 13px;
  }
}

.k-modal-fullscreen-body-header,
.k-modal-fullscreen-body {
  color: var(--KModalFullscreenColor, var(--black-500, color(black-500)));
  padding-left: var(--spacing-lg);
  padding-right: var(--spacing-lg);

  @media only screen and (min-width: ($screen-sm + 1px)) {
    padding-left: 120px;
    padding-right: 120px;
  }

  @media only screen and (min-width: ($screen-md + 1px)) {
    padding-left: 230px;
    padding-right: 230px;
  }
}

.k-modal-fullscreen-body-header {
  padding-top: 0;
  padding-bottom: 0;
}

.k-modal-fullscreen-body {
  padding-bottom: var(--spacing-lg);

  @media only screen and (min-width: ($screen-sm + 1px)) {
    padding-bottom: $fullscreen-modal-padding;
  }
}

.k-modal-fullscreen-body-header {
  margin-top: $fullscreen-modal-padding;
  margin-bottom: 54px;

  .body-header {
    font-size: 32px;
    line-height: 32px;
    font-weight: 600;
    margin-bottom: -4px;
  }

  .body-header-description {
    font-weight: 200;
    font-size: 14px;
    line-height: 22px;
    color: var(--grey-600);
    margin-top: var(--spacing-md);
  }
}

.k-modal-fullscreen-body-description h2 {
  border: none;
}

.k-modal-fullscreen.isOpen .k-modal-fullscreen-dialog {
  overflow-y: auto;
}

.header-content {
  display: inline-block;
  margin-top: var(--spacing-xxs, spacing(xxs));
  padding-left: 6px;
  border-left: 1px solid var(--grey-300);
  line-height: 24px;
}

.k-modal-fullscreen-action-buttons {
  margin-left: auto;
}
</style>

<style lang="scss">
.header-icon {
  .kong-icon.kong-icon-kong {
    position: relative;
    top: 1px;
  }
}
</style>
