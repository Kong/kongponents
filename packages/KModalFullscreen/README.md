# @kongponents/kmodalfullscreen

[![](https://img.shields.io/npm/v/@kongponents/kmodalfullscreen.svg?style=flat-square)](https://www.npmjs.com/package/@kongponents/kmodalfullscreen)

```html
<KModalFullscreen :description="'hello world'">
  Hello from a slot
</KModalFullscreen>
```
