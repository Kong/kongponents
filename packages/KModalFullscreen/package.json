{
  "name": "@kongponents/kmodalfullscreen",
  "version": "6.31.2",
  "description": "A full screen Modal",
  "main": "dist/KModalFullscreen.umd.min.js",
  "componentName": "KModalFullscreen",
  "source": "KModalFullscreen.vue",
  "files": [
    "dist",
    "KModalFullscreen.vue"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@kongponents/kbutton": "^6.31.2",
    "@kongponents/kicon": "^6.31.2",
    "@kongponents/styles": "^6.31.2"
  }
}
