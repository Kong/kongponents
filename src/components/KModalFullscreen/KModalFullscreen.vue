<template>
  <div
    v-if="isVisible"
    :aria-label="title"
    aria-modal="true"
    class="k-modal-fullscreen isOpen"
    role="dialog"
    @keyup.enter="proceed"
    @keyup.esc="close"
  >
    <div
      ref="modalBodyContent"
      class="k-modal-fullscreen-dialog"
      :class="{ 'has-footer': $slots['footer-content'] }"
      tabindex="0"
    >
      <div class="k-modal-fullscreen-body-header">
        <div
          v-if="$slots['body-header'] || bodyHeader"
          class="body-header"
        >
          <slot name="body-header">
            {{ bodyHeader }}
          </slot>
        </div>
        <div
          v-if="$slots['body-header-description'] || bodyHeaderDescription"
          class="body-header-description"
        >
          <slot name="body-header-description">
            {{ bodyHeaderDescription }}
          </slot>
        </div>
      </div>

      <div class="k-modal-fullscreen-body">
        <slot name="default" />
      </div>

      <!-- Header at the bottom to allow proper tabindex -->
      <div class="k-modal-fullscreen-header">
        <div
          aria-level="2"
          class="k-modal-fullscreen-header-description"
          role="heading"
        >
          <div class="k-modal-fullscreen-title">
            <span class="header-icon">
              <slot name="header-icon">
                <KIcon :icon="iconString" />
              </slot>
            </span>
            <span class="header-content">
              <slot name="header-content">{{ title }}</slot>
            </span>
          </div>
          <div class="k-modal-fullscreen-action">
            <div class="k-modal-fullscreen-action-buttons">
              <slot name="action-buttons">
                <KButton
                  :appearance="cancelButtonAppearance"
                  class="cancel-button"
                  @click="close"
                >
                  {{ cancelButtonText }}
                </KButton>
                <KButton
                  :appearance="actionButtonAppearance"
                  class="proceed-button"
                  @click="proceed"
                >
                  {{ actionButtonText }}
                </KButton>
              </slot>
            </div>
          </div>
        </div>
      </div>
      <!-- Footer -->
      <div
        v-if="$slots['footer-content']"
        class="k-modal-fullscreen-footer"
      >
        <slot name="footer-content" />
        <div class="k-modal-fullscreen-action">
          <div class="k-modal-fullscreen-action-buttons">
            <slot name="action-buttons">
              <KButton
                :appearance="actionButtonAppearance"
                class="proceed-button"
                @click="proceed"
              >
                {{ actionButtonText }}
              </KButton>
            </slot>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { watch, ref, computed, onMounted, onUnmounted, onBeforeUnmount, nextTick, PropType } from 'vue'
import KButton from '@/components/KButton/KButton.vue'
import KIcon from '@/components/KIcon/KIcon.vue'
import type { ButtonAppearance } from '@/types'

const props = defineProps({
  /**
  * Set the text of the title, if using title slot
  */
  title: {
    type: String,
    required: true,
  },
  /**
  * Set the title in the body
  */
  bodyHeader: {
    type: String,
    default: '',
  },
  /**
  * Text to display as a description of the body's title
  */
  bodyHeaderDescription: {
    type: String,
    default: '',
  },
  /**
  *  Pass whether or not the modal should be visible
  */
  isVisible: {
    type: Boolean,
    default: false,
  },
  /**
  * Set the text of the close/cancel button
  */
  cancelButtonText: {
    type: String,
    default: 'Cancel',
  },
  /**
  * Set the text of the action/proceed button
  */
  actionButtonText: {
    type: String,
    default: 'Save',
  },
  /**
  * Set the appearance of the action/proceed button
  */
  actionButtonAppearance: {
    type: String as PropType<ButtonAppearance>,
    default: 'primary',
  },
  /**
  * Set the appearance of the close/cancel button
  */
  cancelButtonAppearance: {
    type: String as PropType<ButtonAppearance>,
    default: 'outline',
  },
  /**
  *  Pass the type of icon for the header on the left
  */
  iconString: {
    type: String,
    default: 'kong',
  },
})

const emit = defineEmits(['canceled', 'proceed'])

const modalBodyContent = ref(null)
const isOpen = computed(() => {
  return !!props.isVisible
})

watch(() => props.isVisible, async () => {
  if (isOpen.value) {
    document.body.style.overflow = 'hidden'

    await nextTick()

    if (modalBodyContent.value) {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      modalBodyContent.value.focus()
    }
  } else {
    document.body.style.overflow = ''
  }
})

const handleKeydown = (e: any) => {
  if (props.isVisible) {
    if (e.keyCode === 27) { // `esc` key
      close()
    } else if (e.keyCode === 13) { // `enter` key
      proceed()
    }
  }
}

const close = () => {
  emit('canceled')
}

const proceed = () => {
  emit('proceed')
}

onMounted(() => {
  document.addEventListener('keydown', handleKeydown)
})

onBeforeUnmount(() => {
  document.removeEventListener('keydown', handleKeydown)
})

onUnmounted(() => {
  document.body.style.overflow = ''
})
</script>

<style lang="scss" scoped>
@import '@/styles/variables';
@import '@/styles/tmp-variables';
@import '@/styles/functions';
$kmodalfullscreen-viewport-md: 992px;
$fullscreen-modal-padding: 64px;

.k-modal-fullscreen-dialog {
  background: var(--white, var(--kui-color-background, $kui-color-background));
  bottom: var(--kui-space-0, $kui-space-0);
  left: var(--kui-space-0, $kui-space-0);
  padding-top: $fullscreen-modal-padding * 2;
  position: fixed;
  right: var(--kui-space-0, $kui-space-0);
  top: var(--kui-space-0, $kui-space-0);
  width: 100vw;
  z-index: 9999;

  @media (min-width: $kui-breakpoint-phablet) {
    padding-top: $fullscreen-modal-padding;
  }

  &.has-footer {
    padding-bottom: $fullscreen-modal-padding * 2;

    @media (min-width: $kui-breakpoint-phablet) {
      padding-bottom: $fullscreen-modal-padding;
    }
    .k-modal-fullscreen-header {
      position: absolute;
    }
  }
}

.k-modal-fullscreen-header {
  background-color: var(--white, var(--kui-color-background, $kui-color-background));
  border-bottom: var(--kui-border-width-10, $kui-border-width-10) solid var(--grey-300, var(--kui-color-border-neutral-weak, $kui-color-border-neutral-weak));
  display: flex;
  flex-direction: column;
  padding: var(--spacing-lg, var(--kui-space-80, $kui-space-80)) var(--kui-space-0, $kui-space-0);
  position: fixed;
  top: var(--kui-space-0, $kui-space-0);
  width: 100%;
  z-index: 1009;

  .k-modal-fullscreen-header-description {
    color: var(--KModalFullscreenHeaderColor, var(--kui-color-text-neutral-strongest, $kui-color-text-neutral-strongest));
    display: flex;
    flex-direction: column;
    font-size: var(--KModalFullscreenHeaderSize, var(--kui-font-size-60, $kui-font-size-60));
    font-weight: var(--KModalFullscreenHeaderWeight, var(--kui-font-weight-semibold, $kui-font-weight-semibold));
    justify-content: space-between;

    @media (min-width: $kui-breakpoint-phablet) {
      flex-direction: row;
    }
  }
}

.k-modal-fullscreen-footer {
  align-items: center;
  background-color: var(--white, var(--kui-color-background, $kui-color-background));
  border-top: var(--kui-border-width-10, $kui-border-width-10) solid var(--grey-300, var(--kui-color-border-neutral-weak, $kui-color-border-neutral-weak));
  bottom: var(--kui-space-0, $kui-space-0);
  box-shadow: 0px 0px 20px var(black-10, $tmp-color-black-10);
  display: inline-flex;
  justify-content: space-between;
  padding: var(--spacing-lg, var(--kui-space-80, $kui-space-80)) var(--kui-space-0, $kui-space-0);
  padding-left: var(--spacing-xl, var(--kui-space-90, $kui-space-90));
  position: fixed;
  width: 100%;
  z-index: 1009;
}

.k-modal-fullscreen-title {
  display: inline-flex;
  justify-content: center;
  margin-bottom: var(--spacing-xs, var(--kui-space-40, $kui-space-40));
  position: relative;

  @media (min-width: $kui-breakpoint-phablet) {
    justify-content: flex-start;
    margin-bottom: var(--kui-space-0, $kui-space-0);
    margin-left: var(--kui-space-90, $kui-space-90); // This was 36px, switched it to 32px
  }
}

.k-modal-fullscreen-action {
  display: inline-flex;
  justify-content: center;
  margin-left: var(--kui-space-50, $kui-space-50) !important;
  margin-right: var(--spacing-xl, var(--kui-space-90, $kui-space-90));

  & button,
  & :deep(button) {
    font-size: var(--kui-font-size-20, $kui-font-size-20);
    font-weight: var(--kui-font-weight-semibold, $kui-font-weight-semibold);
    height: var(--kui-space-100, $kui-space-100);
    line-height: var(--kui-line-height-10, $kui-line-height-10);
    margin-left: var(--spacing-md, var(--kui-space-60, $kui-space-60));
  }

  @media (min-width: $kui-breakpoint-phablet) {
    justify-content: flex-end;
  }
}

.k-modal-fullscreen-body-header,
.k-modal-fullscreen-body {
  color: var(--KModalFullscreenColor, var(--black-500, var(--kui-color-text, $kui-color-text)));
  padding-left: var(--spacing-lg, var(--kui-space-80, $kui-space-80));
  padding-right: var(--spacing-lg, var(--kui-space-80, $kui-space-80));

  @media (min-width: $kui-breakpoint-phablet) {
    padding-left: 120px;
    padding-right: 120px;
  }

  @media (min-width: $kmodalfullscreen-viewport-md) {
    padding-left: 230px;
    padding-right: 230px;
  }
}

.k-modal-fullscreen-body {
  padding-bottom: var(--spacing-lg, var(--kui-space-80, $kui-space-80));
  @media (min-width: $kui-breakpoint-phablet) {
    padding-bottom: $fullscreen-modal-padding;
  }
}

.k-modal-fullscreen-body-header {
  margin-bottom: var(--spacing-xl, var(--kui-space-90, $kui-space-90));
  margin-top: $fullscreen-modal-padding;
  padding-bottom: var(--kui-space-0, $kui-space-0);
  padding-top: var(--kui-space-0, $kui-space-0);

  .body-header {
    font-size: var(--type-xxxl, var(--kui-line-height-60, $kui-line-height-60));
    font-weight: var(--kui-font-weight-semibold, $kui-font-weight-semibold);
    line-height: var(--type-xxxl, var(--kui-line-height-60, $kui-line-height-60));
    margin-bottom: calc(-1 * var(--kui-space-20, $kui-space-20));
  }

  .body-header-description {
    color: var(--grey-600, var(--kui-color-text-neutral-strong, $kui-color-text-neutral-strong));
    font-size: var(--kui-font-size-30, $kui-font-size-30);
    font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
    line-height: var(--type-xl, var(--kui-line-height-30, $kui-line-height-30));
    margin-top: var(--spacing-md, var(--kui-space-60, $kui-space-60));
  }
}

.k-modal-fullscreen-body-description h2 {
  border: none;
}

.k-modal-fullscreen.isOpen .k-modal-fullscreen-dialog {
  overflow-y: auto;
}

.header-content {
  border-left: var(--kui-border-width-10, $kui-border-width-10) solid var(--grey-300, var(--kui-color-border-neutral-weak, $kui-color-border-neutral-weak));
  display: inline-block;
  line-height: var(--kui-space-80, $kui-space-80);
  margin-bottom: var(--kui-space-auto, $kui-space-auto) !important;
  margin-top: var(--kui-space-auto, $kui-space-auto) !important;
  padding-left: var(--kui-space-30, $kui-space-30);
}

.k-modal-fullscreen-action-buttons {
  button,
  :deep(button) {
    margin-left: var(--spacing-md, var(--kui-space-60, $kui-space-60));
  }

  @media (min-width: $kui-breakpoint-phablet) {
    margin-left: var(--kui-space-auto, $kui-space-auto) !important;
  }
}
</style>

<style lang="scss">
.header-icon {
  margin-bottom: var(--kui-space-auto, $kui-space-auto) !important;
  margin-top: var(--kui-space-auto, $kui-space-auto) !important;
  padding-right: var(--kui-space-40, $kui-space-40) !important;
  .kong-icon.kong-icon-kong {
    position: relative;
    top: 1px;
  }
}
</style>
