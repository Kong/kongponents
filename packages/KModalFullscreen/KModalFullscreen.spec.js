import { mount } from '@vue/test-utils'
import KModalFullscreen from '@/KModalFullscreen/KModalFullscreen'

describe('KModalFullscreen', () => {
  it('renders proper content when using slots', () => {
    const headerIcon = 'This is some header icon text'
    const headerText = 'This is some header text'
    const bodyHeader = 'This is some body header text'
    const bodyHeaderDescription = 'This is some body header description text'
    const bodyText = 'This is some body text'
    const wrapper = mount(KModalFullscreen, {
      propsData: {
        isVisible: true,
        title: headerText
      },
      slots: {
        'header-icon': `<div>${headerIcon}</div>`,
        'header-content': `<div>${headerText}</div>`,
        'body-header': `<div>${bodyHeader}</div>`,
        'body-header-description': `<div>${bodyHeaderDescription}</div>`,
        'body-content': `<div>${bodyText}</div>`
      }
    })

    expect(wrapper.find('.k-modal-fullscreen-title .header-icon').html()).toEqual(expect.stringContaining(headerIcon))
    expect(wrapper.find('.k-modal-fullscreen-header').html()).toEqual(expect.stringContaining(headerText))
    expect(wrapper.find('.k-modal-fullscreen-body-header .body-header').html()).toEqual(expect.stringContaining(bodyHeader))
    expect(wrapper.find('.k-modal-fullscreen-body-header .body-header-description').html()).toEqual(expect.stringContaining(bodyHeaderDescription))
    expect(wrapper.find('.k-modal-fullscreen-body').html()).toEqual(expect.stringContaining(bodyText))
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders proper content when using action-buttons slot', () => {
    const actionButtonsText = 'This is some action buttons text'
    const wrapper = mount(KModalFullscreen, {
      propsData: {
        isVisible: true,
        title: 'Test Me'
      },
      slots: {
        'action-buttons': `<div>${actionButtonsText}</div>`
      }
    })

    expect(wrapper.find('.k-modal-fullscreen-action').html()).toEqual(expect.stringContaining(actionButtonsText))
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders proper content when using props', () => {
    const title = 'Sweet prop title'
    const actionButtonText = 'Sweet prop actionButton'
    const cancelButtonText = 'Sweet prop cancelButton'
    const bodyHeader = 'Sweet prop bodyHeader'
    const bodyHeaderDescription = 'Sweet prop bodyHeaderDescription'
    const content = 'Sweet prop content'
    const wrapper = mount(KModalFullscreen, {
      propsData: {
        isVisible: true,
        title,
        actionButtonText,
        cancelButtonText,
        bodyHeader,
        bodyHeaderDescription,
        content,
        iconString: 'immunity'
      }
    })

    expect(wrapper.find('.k-modal-fullscreen-title .header-icon .kong-icon-immunity').exists()).toBeTruthy()
    expect(wrapper.find('.cancel-button').html()).toEqual(expect.stringContaining(cancelButtonText))
    expect(wrapper.find('.proceed-button').html()).toEqual(expect.stringContaining(actionButtonText))
    expect(wrapper.find('.k-modal-fullscreen-header').html()).toEqual(expect.stringContaining(title))
    expect(wrapper.find('.k-modal-fullscreen-body-header .body-header').html()).toEqual(expect.stringContaining(bodyHeader))
    expect(wrapper.find('.k-modal-fullscreen-body-header .body-header-description').html()).toEqual(expect.stringContaining(bodyHeaderDescription))
    expect(wrapper.find('.k-modal-fullscreen-body').html()).toEqual(expect.stringContaining(content))
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('emits close when hitting escape', () => {
    const wrapper = mount(KModalFullscreen, {
      propsData: {
        title: 'Test Me',
        isVisible: true
      },
      attachToDocument: true
    })

    wrapper.find('.k-modal-fullscreen').trigger('keydown.esc')
    expect(wrapper.emitted().canceled).toHaveLength(1)
  })

  it('tears down event listeners', () => {
    const AEL = jest.fn()
    const REL = jest.fn()

    window.document.addEventListener = AEL
    window.document.removeEventListener = REL

    const wrapper = mount(KModalFullscreen, {
      propsData: {
        title: 'Test Me'
      },
      attachToDocument: true
    })

    wrapper.destroy()
    expect(AEL).toHaveBeenCalledTimes(1)
    expect(REL).toHaveBeenCalledTimes(1)
  })

  it('matches snapshot', () => {
    const wrapper = mount(KModalFullscreen, {
      propsData: {
        title: 'Test Me'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
