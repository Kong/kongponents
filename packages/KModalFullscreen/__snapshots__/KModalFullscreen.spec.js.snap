// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KModalFullscreen matches snapshot 1`] = `undefined`;

exports[`KModalFullscreen renders proper content when using action-buttons slot 1`] = `
<div aria-label="Test Me" role="dialog" aria-modal="true" class="k-modal-fullscreen isOpen">
  <div class="k-modal-dialog">
    <div class="k-modal-header">
      <div role="heading" aria-level="2" class="k-modal-header-description mb-5">
        <div class="k-modal-title"><span class="header-icon"><span class="kong-icon mr-2 kong-icon-kong"><svg viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Kong</title>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m16.28 36.66 1-1.3h7.45l3.88 4.96-.7 1.68h-9.6l.24-1.68-2.27-3.66Z" fill="#169FCC"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m18.55 19.75 3.6-6.21h4.19L45.1 35.35 43.65 42H35.6l.5-1.87-17.55-20.38Z" fill="#14B59A"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m22.92 12.36 1.72-3.19L29.8 5l8.85 6.94-1.15 1.17 1.54 2.13v2.28l-4.4 3.6-7.4-8.76h-4.32Z" fill="#1BC263"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M9.25 26.23h2.33l6.1-5.1 8.08 9.4-2.28 3.41h-7.46l-5.15 6.55L9.7 42H3v-8.03l6.25-7.74Z" fill="#16BDCC"></path>
</svg>
</span></span> <span class="header-content">Test Me</span></div>
        <div class="k-modal-action ml-3">
          <div>This is some action buttons text</div>
        </div>
      </div>
      <div>
        <hr>
      </div>
    </div>
    <div class="k-modal-body-description"></div>
    <div class="k-modal-body"></div>
  </div>
</div>
`;

exports[`KModalFullscreen renders proper content when using props 1`] = `
<div aria-label="Sweet prop title" role="dialog" aria-modal="true" class="k-modal-fullscreen isOpen">
  <div class="k-modal-dialog">
    <div class="k-modal-header">
      <div role="heading" aria-level="2" class="k-modal-header-description mb-5">
        <div class="k-modal-title"><span class="header-icon"><span class="kong-icon mr-2 kong-icon-kong"><svg viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Kong</title>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m16.28 36.66 1-1.3h7.45l3.88 4.96-.7 1.68h-9.6l.24-1.68-2.27-3.66Z" fill="#169FCC"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m18.55 19.75 3.6-6.21h4.19L45.1 35.35 43.65 42H35.6l.5-1.87-17.55-20.38Z" fill="#14B59A"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m22.92 12.36 1.72-3.19L29.8 5l8.85 6.94-1.15 1.17 1.54 2.13v2.28l-4.4 3.6-7.4-8.76h-4.32Z" fill="#1BC263"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M9.25 26.23h2.33l6.1-5.1 8.08 9.4-2.28 3.41h-7.46l-5.15 6.55L9.7 42H3v-8.03l6.25-7.74Z" fill="#16BDCC"></path>
</svg>
</span></span> <span class="header-content">Sweet prop title</span></div>
        <div class="k-modal-action ml-3"><button type="button" class="k-button  mr-2 medium rounded outline">
            Cancel
            <!----></button> <button type="button" class="k-button medium rounded primary">
            Save
            <!----></button></div>
      </div>
      <div>
        <hr>
      </div>
    </div>
    <div class="k-modal-body-description"></div>
    <div class="k-modal-body">Sweet prop content</div>
  </div>
</div>
`;

exports[`KModalFullscreen renders proper content when using slots 1`] = `
<div aria-label="This is some header text" role="dialog" aria-modal="true" class="k-modal-fullscreen isOpen">
  <div class="k-modal-dialog">
    <div class="k-modal-header">
      <div role="heading" aria-level="2" class="k-modal-header-description mb-5">
        <div class="k-modal-title"><span class="header-icon"><span class="kong-icon mr-2 kong-icon-kong"><svg viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Kong</title>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m16.28 36.66 1-1.3h7.45l3.88 4.96-.7 1.68h-9.6l.24-1.68-2.27-3.66Z" fill="#169FCC"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m18.55 19.75 3.6-6.21h4.19L45.1 35.35 43.65 42H35.6l.5-1.87-17.55-20.38Z" fill="#14B59A"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="m22.92 12.36 1.72-3.19L29.8 5l8.85 6.94-1.15 1.17 1.54 2.13v2.28l-4.4 3.6-7.4-8.76h-4.32Z" fill="#1BC263"></path>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M9.25 26.23h2.33l6.1-5.1 8.08 9.4-2.28 3.41h-7.46l-5.15 6.55L9.7 42H3v-8.03l6.25-7.74Z" fill="#16BDCC"></path>
</svg>
</span></span> <span class="header-content"><div>This is some header text</div></span></div>
        <div class="k-modal-action ml-3"><button type="button" class="k-button  mr-2 medium rounded outline">
            Cancel
            <!----></button> <button type="button" class="k-button medium rounded primary">
            Save
            <!----></button></div>
      </div>
      <div>
        <hr>
      </div>
    </div>
    <div class="k-modal-body-description"></div>
    <div class="k-modal-body">
      <div>This is some body text</div>
    </div>
  </div>
</div>
`;
